pwiz_tools/Skyline/Model/Lib/ChromLib/ChromatogramLibrary.cs

```diff
@@ -91,19 +92,35 @@ protected override LibraryChromGroup ReadChromatogram(ChromLibSpectrumInfo info)
                     var timeIntensities = precursor.ChromatogramData;
                     double height = 0;
                     var chromDatas = new List<LibraryChromGroup.ChromData>();
+                    var precursor13 = precursor as Precursor.Format1Dot3;
                     foreach (var transition in precursor.Transitions)
                     {
+                        var transition13 = transition as Data.Transition.Format1Dot3;
+                        var ionType = Helpers.ParseEnum(transition.FragmentType, IonType.y);
+                        IonMobilityValue ionMobility = null;
+                        var im = (ionType == IonType.precursor)
+                            ? precursor13?.IonMobilityMS1
+                            : precursor13?.IonMobilityFragment;
+                        if (im.HasValue)
+                        {
+                            var imUnits = Helpers.ParseEnum(precursor13.IonMobilityType, eIonMobilityUnits.none);
+                            if (imUnits != eIonMobilityUnits.none)
+                            {
+                                ionMobility = IonMobilityValue.GetIonMobilityValue(im, imUnits);
+                            }
+                        }
                         chromDatas.Add(new LibraryChromGroup.ChromData
                             {
                                 Mz = transition.Mz,
                                 Height = transition.Height,
                                 Intensities = timeIntensities.Intensities[transition.ChromatogramIndex],
-                                Charge = transition.Charge == 0 ? precursor.Charge : transition.Charge,

```

Comment:
I think the removal of this line is causing the SplitGraphTest failure.
In older .clib files, the Charge column is sometimes Null for some transitions, in which case it's supposed to use the charge from the precursor.

Reply:
Thanks!